workflow/executor/executor.go

```diff
@@ -947,8 +947,10 @@ func unpack(srcPath string, destPath string, decompressor func(string, string) e
 	if err != nil {
 		return errors.InternalWrapError(err)
 	}
-	if err = decompressor(srcPath, tmpDir); err != nil {
-		return err
+	if decompressor != nil {
```

Comment:
minor: if you had a `noopDecompressor` you could always invoke it safely

Reply:
not exactly sure what you mean by a `noopDecompressor` in this case - do you have an example of somewhere else in the codebase where this pattern is used?

Reply:
`hydrator.Noop` `sso.NullSSO`

Reply:
I had a look at those examples and tried to come up with something similar (below). Not sure if this is what you had in mind, but with the below implementation, there is still nothing that stops the user from just calling the function with `nil`.

```golang
type Decompressor interface {
    Decompress(src string, dest string) error
}

type NoopDecompressor struct {}
func (nd *NoopDecompressor) Decompress(src string, dest string) error {
    return nil
}

type TarDecompressor struct {}
func (td *TarDecompressor) Decompress(src string, dest string) error {
    return common.RunCommand("tar", "-xf", src, "-C", dest)
}

type ZipDecompressor struct {}
func (zd *ZipDecompressor) Decompress(src string, dest string) error {
    // ...
}

...

func unpack(srcPath string, destPath string, decompressor Decompressor) error {
    // ...
    decompressor.Decompress(srcPath, destPath) // could still be nil
    // ...
}

unpack(zipPath, destPath, &ZipDecompressor{})
unpack(tarPath, destPath, &TarDecompressor{})
unpack(srcPath, destPath, &NoopDecompressor{}) // will not be called anywhere at present
unpack(srcPath, destPath, nil) // still compiles, but throws error at runtime
```

Sorry if I misunderstood your message (I am new to Golang), but for me this seems like a lot of code to bypass a simple nil-check :)